File: backend/app/models/seat.py
import uuid
from datetime import datetime

from sqlalchemy import String, DateTime, ForeignKey, Enum as SAEnum
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Seat(Base):
    __tablename__ = "seats"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    event_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("events.id"), nullable=False)
    section: Mapped[str] = mapped_column(String(64), nullable=False)
    row: Mapped[str] = mapped_column(String(16), nullable=False)
    number: Mapped[str] = mapped_column(String(16), nullable=False)
    seat_label: Mapped[str] = mapped_column(String(128), nullable=False)
    status: Mapped[str] = mapped_column(
        SAEnum("available", "reserved", "occupied", "disabled", name="seat_status"),
        default="available",
        nullable=False,
    )
    order_id: Mapped[uuid.UUID | None] = mapped_column(UUID(as_uuid=True), ForeignKey("orders.id"), nullable=True)
    ticket_type_id: Mapped[uuid.UUID | None] = mapped_column(UUID(as_uuid=True), ForeignKey("ticket_types.id"), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    event = relationship("Event", lazy="selectin")
    order = relationship("Order", lazy="selectin")
    ticket_type = relationship("TicketType", lazy="selectin")

File: backend/TicketDesk.Domain/Entities/Seat.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using TicketDesk.Domain.Enums;

namespace TicketDesk.Domain.Entities;

public class Seat
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public Guid SeatMapId { get; set; }

    [Required]
    public string Row { get; set; } = string.Empty;

    [Required]
    public string Number { get; set; } = string.Empty;

    public string? Section { get; set; }

    public SeatStatus Status { get; set; }

    public decimal Price { get; set; }

    public Guid? TicketTypeId { get; set; }

    [ForeignKey(nameof(SeatMapId))]
    public SeatMap SeatMap { get; set; } = null!;

    [ForeignKey(nameof(TicketTypeId))]
    public TicketType? TicketType { get; set; }

    public ICollection<Order> Orders { get; set; } = new List<Order>();
}
